ci/test/lint-docs.sh

```diff
@@ -12,8 +12,17 @@
 # lint.sh — checks for broken links and other HTML errors.
 
 set -euo pipefail
-
+# Save current branch so we can temp checkout main and switch back
+export CURRENT_BRANCH=(git rev-parse --abbrev-ref HEAD)
```

Comment:
I think this wants to be `current_branch=$(git ...)`, but also no need to switch back after you've switched to `main`; Buildkite doesn't care if you futz with Git after you build has started; I think you can just ditch this and L16 and it will all work out.